### PR TITLE
Fix? contributor test trigger

### PR DESCRIPTION
**Pull request by pmeier**
## What does this implement/fix?

As seen in #1711 neither my maintainer status nor @iameskild's comment triggered the contributor tests. However, they are required by our branch protection rules. Meaning, #1711 is effectively stuck right now and cannot be merged. 

As explained in https://github.com/nebari-dev/nebari/pull/1734#issuecomment-1517695618, it is a pain to debug workflows that trigger on comments since they don't use the workflow file from the current PR, but only the one from the default branch. Thus, we probably need to merge this PR and try afterwards. I did that in my fork and it worked.

This PR does two things:

1. Instead of having one large condition, it is split in multiple atomic ones. Since we `&&` them together anyway, this can be trivially achieved.
2. Instead of having an `if:` trigger on multiple jobs, we define one "pre-job" and require that on all others through the `needs:` key. That makes it a lot easier if we ever want to change the condition.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

**Comment by pmeier** [comment]
random comment

**Comment by pmeier** [comment]
Ok, this will be a little tricky to debug, since 

https://github.com/nebari-dev/nebari/blob/4ac4693f90d16117713bd7e84bbb976b030528c2/.github/workflows/contributor.yaml#L3-L5

will only run against the latest commit of the default branch. Meaning, regardless of what changes we make in the PR in this file, they will not be used until after the PR is merged. I'll debug without the comment trigger, but we might have to merge a dummy PR and revert later if it doesn't do what we want it to do.

File: .github/workflows/contributor.yaml [hunk 1]
```diff
@@ -31,25 +47,25 @@ jobs:
           reactions: 'rocket'
 
   kubernetes-tests:
+    needs: trigger-contributor-tests
     name: "Contributor #${{ github.event.issue.number }}"
-    if: ${{ github.event.issue.pull_request && contains(github.event.comment.body, '/bot run tests') && contains(fromJson('["OWNER", "COLLABORATOR", "MEMBER"]'), github.event.comment.author_association)}}
-    uses: nebari-dev/nebari/.github/workflows/kubernetes_test.yaml@release/2022.11.1
+    uses: nebari-dev/nebari/.github/workflows/kubernetes_test.yaml@release/2023.4.1
```
Comment (pmeier) [inline .github/workflows/contributor.yaml:52]: Driveby. We should probably add this to our release checklist.

Comment (pmeier) [reply .github/workflows/contributor.yaml:52]: One second thought: we do we want to run the "released" tests at all. Shouldn't this be `@develop`? Otherwise no changes to the default branch will be considered when running contributor tests.

Comment (iameskild) [reply .github/workflows/contributor.yaml:52]: @pmeier I agree, `@develop` is what we want.

**Comment by iameskild** [comment]
@pmeier do you want to work together on this? It would be great to get this working since we have new contributors opening PRs :)

**Comment by pmeier** [comment]
Feel free to push to my branch

**Review by iameskild — APPROVED** [review]
As @pmeier pointed out in the PR description, to debug we need to merge this. As far as I can tell, this won't have any impact on the rest of the codebase/repo.
